docs/developer_guide/example_docstrings.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import annotations
+
+from typing import Any, Dict, Iterator
+
+
+def example_function_args(
+    param1: int,
+    param2: str | int,
+    optional_param: list[str] | None = None,
```

Comment:
We are using the List typing from "typing" in the whole repository, instead of the native list from python. Maybe it would be nice to share that across the codebase.